src/stdune/hashtbl.ml

```diff
@@ -58,6 +58,11 @@ module Make(H : Hashable.S) = struct
     | Result.Ok h -> h
     | Error (_, _, _) ->
       Code_error.raise "Hashtbl.of_list_exn duplicate keys" []
+
+  let add_exn t key data =
+    match find t key with
+    | None -> set t key data
+    | Some _ -> Code_error.raise "Hastbl.add_exn: key already exists" []
```

Comment:
Not showing the key here makes for very poor error messages, and you're removing some of the existing better error messages too.

Maybe we should keep the custom error messages for now?